doc/tutorial/02_index/index.md

```diff
@@ -1,19 +1,274 @@
+[<b>Interleaved Bloom Filter (IBF)</b>](https://docs.seqan.de/seqan/3-master-user/classseqan3_1_1interleaved__bloom__filter.html#details)\,
+\ref raptor::hierarchical_interleaved_bloom_filter "Hierarchical Interleaved Bloom Filter (HIBF)"}
 
 [TOC]
 
-# Index of large collections of nucleotide sequences
+# Indexing large collections of nucleotide sequences
 
-ToDo ...
+## A first index
 
-## EXAMPLES
+Raptor consists of two methods, `build` and `search`. The former creates an index over a given dataset so that it can be
+searched efficiently with `raptor search`.
+In this tutorial, we will look at `raptor build` in more detail.
 
-`raptor build --kmer 19 --window 23 --size 8m --output raptor.index all_bin_paths.txt`
+Raptor can be used as a pre-filter for applications where searching the complete dataset is not feasible. For example,
+in read mapping you might want to compare your genome to the genome of 100'000 other people.
+It can also be used for metagenomic classification, i.e., which microbes are in a single sample.
 
-`raptor build --kmer 19 --window 23 --compute-minimiser --output precomputed_minimisers all_bin_paths.txt`
+\see Related tools:
+- Rna-seq expression analysis: [Needle](https://www.seqan.de/apps/needle.html), a tool for storing sequencing
+experiments in such a way that approximate quantification of large sequencing data sets is possible.
+- Metagenomics: [Ganon/Ganon2](https://github.com/pirovc/ganon), a tool for comparative metagenomics analysis in short
+time using the whole of the quickly growing number of assembled sequences openly available.
 
-`raptor build --size 8m --output minimiser_raptor.index all_minimiser_paths.txt`
+Regardless of the application, we start with a data set of different nucleotide sequences over which we want to build an
+index.
+
+These sequences are typically available as
+[FASTA](https://docs.seqan.de/seqan/3-master-user/classseqan3_1_1format__fasta.html#a86790afd92e0229cccbc20be20d5758a) or
+[FASTQ](https://docs.seqan.de/seqan/3-master-user/classseqan3_1_1format__fastq.html#details) files, but can be in any
+format supported by
+[`seqan3::sequence_file_input`](https://docs.seqan.de/seqan/3-master-user/tutorial_sequence_file.html).
+
+\assignment{Assignment 1: Create example files and index them}
+Copy and paste the following FASTA files to some location, e.g. the tmp directory:
+
+mini_1.fasta:
+```fasta
+>chr1
+AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACGCGTTCATTAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+```
+mini_2.fasta:
+```fasta
+>chr2
+AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACGCGTCATTAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+```
+mini_3.fasta:
+```fasta
+>chr3
+AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+```
+and then create an all_paths.txt and run raptor build.
+\endassignment
+
+\solution
+Your all_paths.txt should look like:
+```txt
+mini_1.fasta
+mini_2.fasta
+mini_3.fasta
+```
```

Comment:
You could add a note afterwards that it might be preferable to store the absolute path